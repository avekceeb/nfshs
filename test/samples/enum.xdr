/*
 * Types of files:
 */
enum filekind {
   TEXT = 0,       /* ascii data */
   DATA = 1,       /* raw data   */
   EXEC = 2        /* executable */
};
