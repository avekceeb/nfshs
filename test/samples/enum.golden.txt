( [ Left
      (TypeDefEnum
         (Identifier "filekind")
         (( Identifier "TEXT" , Left (DecConstant 0) ) :|
            [ ( Identifier "DATA" , Left (DecConstant 1) )
            , ( Identifier "EXEC" , Left (DecConstant 2) )
            ]))
  ]
, ParserState
    []
    [ ( SourcePos
          { sourceName = "test/samples/enum.xdr"
          , sourceLine = Pos 4
          , sourceColumn = Pos 1
          }
      , ( Identifier "filekind"
        , TypeEnum
            (( Identifier "TEXT" , Left (DecConstant 0) ) :|
               [ ( Identifier "DATA" , Left (DecConstant 1) )
               , ( Identifier "EXEC" , Left (DecConstant 2) )
               ])
        )
      )
    ]
    (fromList [] :| [])
)